src/main/kotlin/com/devfloor/untitled/article/application/ArticleFacade.kt

```diff
@@ -12,8 +16,23 @@ class ArticleFacade(
     private val articleService: ArticleService,
     private val articleHashtagService: ArticleHashtagService,
     private val articleFavoriteService: ArticleFavoriteService,
-    private val articleOptionService: ArticleOptionService
+    private val articleOptionService: ArticleOptionService,
+    private val optionService: OptionService,
+    private val hashtagFacade: HashtagFacade
 ) {
+    @Transactional
+    fun create(request: ArticleRequest, user: User): Long {
+        val article = articleService.create(request.toArticle(request, user))
+        if (request.isOptionsNotEmpty()) {
+            optionService.showAllByOptionType(request.options)
+                .map { ArticleOption(article, it) }
+                .let { articleOptionService.createAll(it) }
+        }
+        val hashtags = hashtagFacade.createAll(request.hashtags)
+        articleHashtagService.createAll(hashtags.map { ArticleHashtag(article, it) })
```

Comment:
Hashtag를 생성(혹은 조회)하고 나서는 항상 ArticleHashtag도 함께 생성해야 할 것 같은데,
여기 혹시 이렇게 바꿔 두 로직을 HashtagFacade 내부의 한 메서드에 묶어놓으면 어떨까요?

```suggestion
        request.hashtags.forEach { hashtagFacade.createByName(article, it) }
```

아래는 HashtagFacade에서의 `createByName` 입니다.
```
    @Transactional
    fun createByName(article: Article, hashtagName: String) {
        val hashtag = if (!hashtagService.existsByName(hashtagName))
            hashtagService.create(Hashtag(hashtagName))
        else hashtagService.showByName(hashtagName)
        articleHashtagService.create(ArticleHashtag(article, hashtag))
    }
```

Reply:
그리고 해당 메소드에서 항상 Hashtag를 create하지는 않는데, 혹시 더 좋은 메서드명 없을까요?

Reply:
이건 나도 계속 고민이네.. 어떻게 하는게 더 좋을지..